src/pages/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,134 @@
+import React, { useState } from 'react';
+import { useFormik } from 'formik';
+import * as Yup from 'yup';
+import {
+  TextField,
+  IconButton,
+  List,
+  ListItem,
+  ListItemText,
+} from '@material-ui/core';
+import AddIcon from '@material-ui/icons/Add';
+import DeleteIcon from '@material-ui/icons/Delete';
+
+interface TodoItem {
+  title: string;
+  id: number;
+  done: boolean;
+}
+
+interface MyFormValues {
+  title: string;
+}
+
+const TodoList: React.FC = () => {
+  const [list, setList] = useState<TodoItem[]>([]);
+  const [itemFilter, setItemFilter] = useState('');
+  const initialValues: MyFormValues = { title: '' };
+  const formik = useFormik({
+    initialValues,
+    onSubmit: (values, actions) => {
+      const newTodo: TodoItem = {
+        title: values.title,
+        id: Math.random(),
+        done: false,
+      };
+      setList((prev) => [newTodo, ...prev]);
+      actions.resetForm();
+    },
+    validationSchema: Yup.object().shape({
+      title: Yup.string()
+        .required('Campo obrigatório')
+        .max(20, 'O limite máximo é de 20 caracteres'),
+    }),
+  });
+
+  const handleDelete = (id: TodoItem['id']) => {
+    setList((prev) => prev.filter((item) => item.id !== id));
+  };
+
+  const handleFilter = (title: string) => {
+    if (title.length === 0) {

```

Comment:
Can you apply clean code here?

```suggestion
    const isTitleEmpty = title.length === 0;
    if (isTitleEmpty) {

```